perfmetrics/scripts/ls_metrics/listing_benchmark_test.py

```diff
@@ -0,0 +1,519 @@
+                                        'file_size': '1kb',
+                                        'num_folders': 1,
+                                        'folders': [
+                                            {
+                                                'name': 'subsubdir1',
+                                                'num_files': 1,
+                                                'file_name_prefix': 'file',
+                                                'file_size': '1kb'
+                                            }
+                                        ]
+                            }
+                        ]
+            },
+        {
+                'name': '2KB_3files_1subdir',
+                'num_files': 3,
+                'file_name_prefix': 'file',
+                        'file_size': '2kb',
+                        'num_folders': 1,
+                        'folders': [
+                            {
+                                'name': 'subdir1'
+                            }
+                        ]
+                },
+        {
+                'name': '1KB_1files_0subdir',
+                'num_files': 1,
+                'file_size': '1kb',
+                'file_name_prefix': 'file'
+                }
+    ]
+}
+
+METRICS1 = [1.234, 0.995, 0.121, 0.222, 0.01709]
+METRICS2 = [90.45, 1.95, 0.334, 7.090, 0.001]
+METRICS3 = [100, 7, 6, 51, 21]
+
+DIRECTORY_STRUCTURE1 = ParseDict(
+    DIRECTORY_STRUCTURE1, directory_proto.Directory())
+DIRECTORY_STRUCTURE2 = ParseDict(
+    DIRECTORY_STRUCTURE2, directory_proto.Directory())
+DIRECTORY_STRUCTURE3 = ParseDict(
+    DIRECTORY_STRUCTURE3, directory_proto.Directory())
+
+
+class ListingBenchmarkTest(unittest.TestCase):
+
+    def test_parse_results_type1(self):
+        metrics = listing_benchmark._parse_results(
+            DIRECTORY_STRUCTURE1.folders, {}, 'fake_test', 5)
+        self.assertEqual(metrics, {})
+
+    def test_parse_results_type2(self):
+        metrics = listing_benchmark._parse_results(DIRECTORY_STRUCTURE2.folders, {
+            '2KB_3files_0subdir': METRICS1,
+            '1KB_2files_0subdir': METRICS2,
+            '1KB_0files_0subdir': METRICS3
+        }, 'fake_test', 5)
+
+        self.assertEqual(metrics,
+                         {
+                             '2KB_3files_0subdir':
+                             {
+                                 'Test Desc.': 'fake_test',
+                                 'Number of samples': 5,
+                                 'Mean': 0.517818,
+                                 'Median': 0.222,
+                                 'Standard Dev': 0.5559497869592182,
+                                 'Quantiles':
+                                 {
+                                     '0 %ile': 0.01709,
+                                     '20 %ile': 0.100218,
+                                     '40 %ile': 0.1816,
+                                     '60 %ile': 0.5311999999999999,
```

Comment:
Can we have the code do round off to 3 decimal places?

Reply:
rounded the numbers to 3 decimal places.